### PR TITLE
Store toggle status of the user profile rank graph

### LINKED ISSUE #3967: profile: remember last setting for collapsed/expanded ranking info
not much point if you have to click it every single time you visit someone's profile to collapse the ranking area
image

### PR DESCRIPTION
**Pull request by LiquidPL**
Fixes #3967.

### CODE REVIEW

**Comment by LiquidPL** [comment]
Also relevant to #4434 but I'm gonna leave this open as it related to any other potential toggles that need this done too.

File: app/Models/UserProfileCustomization.php [hunk 1]
```diff
@@ -121,6 +121,16 @@ public function setExtrasOrderAttribute($value)
...
     }
 
+    public function getRankingExtendedAttribute()
+    {
+        return $this->options['ranking_extended'] ?? true;
+    }
+
+    public function setRankingExtendedAttribute($value)
+    {
+        $this->setOption('ranking_extended', $value === 'true');
```
Comment (notbakaneko) [inline app/Models/UserProfileCustomization.php]: so `'true'` is `true` but `true` itself is not `true`?

File: resources/assets/coffee/react/profile-page/detail.coffee [hunk 2]
```diff
@@ -80,4 +80,10 @@ export class Detail extends React.PureComponent
 
 
   toggleExtend: =>
+    $.ajax laroute.route('account.options'),
+      method: 'put',
+      dataType: 'json',
+      data:
+        user_profile_customization: ranking_extended: !@state.extended
```
Comment (notbakaneko) [inline resources/assets/coffee/react/profile-page/detail.coffee]: I: don't: recommend: doing: this

File: app/Http/Controllers/AccountController.php [hunk 3]
```diff
@@ -177,6 +177,7 @@ public function updateOptions()
         $params = get_params(request(), 'user_profile_customization', [
             'comments_sort:string',
             'extras_order:string[]',
+            'ranking_extended:string',
```
Comment (notbakaneko) [inline app/Http/Controllers/AccountController.php]: I think this should be `rankings_expanded`

File: resources/assets/coffee/react/profile-page/detail.coffee [hunk 4]
```diff
@@ -32,7 +32,7 @@ export class Detail extends React.PureComponent
   constructor: (props) ->
     super props
 
-    @state = extended: true
+    @state = extended: props.user.ranking_extended
```
Comment (notbakaneko) [inline resources/assets/coffee/react/profile-page/detail.coffee]: if it's simple enough, rename `extended` to `expanded` to be inline with the naming for expanded elements elsewhere

File: app/Transformers/UserTransformer.php [hunk 5]
```diff
@@ -96,6 +96,7 @@ public function transform(User $user)
             'post_count' => $user->user_posts,
             'profile_colour' => $user->user_colour,
             'profile_order' => $profileCustomization->extras_order,
+            'ranking_expanded' => $profileCustomization->ranking_expanded,
```
Comment (notbakaneko) [inline app/Transformers/UserTransformer.php]: should be using the preferences of the user viewing the page, not the user of the page viewed

File: app/Transformers/UserTransformer.php [hunk 6]
```diff
@@ -50,6 +50,7 @@ class UserTransformer extends Fractal\TransformerAbstract
         'unranked_beatmapset_count',
         'unread_pm_count',
         'user_achievements',
+        'ranking_expanded',
```
Comment (notbakaneko) [inline app/Transformers/UserTransformer.php]: something generic like 'user_preferences' since it should be able to have multiple settings under the key

**Pull request merged**